### ISSUE TITLE
List - `onOrder` has accessibility issues with aria roles and `<li>` not being contained by `<ul>` or `<ol>`

### ISSUE THREAD

**Issue by jcfilben** [grommet/grommet#7425]
### Expected Behavior
List with `onOrder` should pass storybook's automated accessibility tests

### Actual Behavior
The List/Order story has accessibility violations

### URL, screen shot, or Codepen exhibiting the issue
See the List/Order story accessibility tab:
https://storybook.grommet.io/?path=/story/visualizations-list-order--order

<img width="507" alt="Screenshot 2024-11-08 at 3 18 23 PM" src="https://github.com/user-attachments/assets/2a9699c2-d619-4c46-b08e-f93e7c4288fa">

### Steps to Reproduce

1. 2. 3.

### Your Environment

- Grommet version:
- Browser Name and version:
- Operating System and version (desktop or mobile):
